pkg/utils/parse_arf_result.go

```diff
@@ -717,54 +717,89 @@ func toArrayByComma(format string) []string {
 func parseValues(remContent string, resultValues map[string]string) (string, []string, []string, error) {
 	var valuesUsedList []string
 	var valuesMissingList []string
-	var valuesParsedList []string
+
 	//find everything start and end with {{}}
 	re := regexp.MustCompile(`\{\{[^}]*\}\}`)
 	contentList := re.FindAllString(remContent, -1)
 	fixedText := remContent
 	if len(contentList) == 0 {
 		return remContent, valuesUsedList, valuesMissingList, nil
-	} // no processing needed, no urlencoded data
-	//process urlencoded content file data one by one and replace them in fixedText
+	}
+	// there are two types of content we need to process, one is url-encoded machine config source ex. {{ -a%20always0-F%20di }},
+	// the other one is not url_encoded ex. {{.var_some_value}}, we are going to take care of url-encoded content first, and then
+	// feed the processed content to template again
+
 	for _, content := range contentList {
-		//take out `{{ `,' }}' from content
+		// if third character is not space, skip non url-encoded content
+		if content[2:3] != " " {
+			continue
+		}
+		// take out `{{ `,' }}' out from content
 		trimmedContent := content[3:][:len(content)-6]
-		decodeContent, decodeErr := url.QueryUnescape(trimmedContent)
+		// trimmedContent here should only contain url-encoded content, check if it contains illegall character space
```

Comment:
now that I read this again:
```
		trimmedContent := content[3:][:len(content)-6]
```
I'm confused (again)... could we turn the `3` and `len(content)-6` into variables instead to make them more descriptive? If needed, let's add a bit more specific comments on why those numbers were chosen.

the `3` is not too hard, as it's clearly `{{ `, which is 3 characters. But the length  `- 6` is a little trickier and I don't quite remember why it was done this way. Since it's something we might easily forget (which would affect maintenance) being more descriptive is necessary. [1]

[1] https://dev.to/tbutterwith/avoid-magic-numbers-to-increase-code-readability-1da1

Reply:
That's true! But now I thought about this, someone who write content might unintentionally bring a extra space might crash the remediation

Reply:
> That's true! But now I thought about this, someone who write content might unintentionally bring a extra space might crash the remediation

sounds to me like this would be a nice unit-test to have here!

Reply:
> now that I read this again:
> 
> ```
> 		trimmedContent := content[3:][:len(content)-6]
> ```
> 
> I'm confused (again)... could we turn the `3` and `len(content)-6` into variables instead to make them more descriptive? If needed, let's add a bit more specific comments on why those numbers were chosen.
> 
> the `3` is not too hard, as it's clearly `{{ `, which is 3 characters. But the length `- 6` is a little trickier and I don't quite remember why it was done this way. Since it's something we might easily forget (which would affect maintenance) being more descriptive is necessary. [1]
> 
> [1] https://dev.to/tbutterwith/avoid-magic-numbers-to-increase-code-readability-1da1

- 6 is because when we use len(content), the real length already become len(content)-3 when we trimmed the first three character, tha's why we need to do -6 to trim the last three character.